diff --git a/arch/arm64/boot/dts/mediatek/mt8370-genio-510-evk.dts b/arch/arm64/boot/dts/mediatek/mt8370-genio-510-evk.dts
index d17008d677b7..c04ba4061718 100644
--- a/arch/arm64/boot/dts/mediatek/mt8370-genio-510-evk.dts
+++ b/arch/arm64/boot/dts/mediatek/mt8370-genio-510-evk.dts
@@ -27,18 +27,21 @@ aliases {
 		ethernet0 = &eth;
 		serial0 = &uart0;
 		dsi0 = &dsi0;
+		rtc0 = &pcf85063;  /* 20250905 */
+		rtc1 = &mt6359rtc; /* 20250905 */
 	};
 
 	backlight_lcd0: backlight-lcd0 {
 		compatible = "pwm-backlight";
-		pwms = <&disp_pwm0 0 500000>;
-		enable-gpios = <&pio 8 GPIO_ACTIVE_HIGH>;
+		pwms = <&disp_pwm1 0 500000>; /* 20250905 */
+		enable-gpios = <&pio 45 GPIO_ACTIVE_HIGH>; /* 20250905 */
 		brightness-levels = <0 1023>;
 		num-interpolated-steps = <1023>;
 		default-brightness-level = <576>;
-		status = "disabled";
+		status = "okay"; /* 20250905 */
 	};
 
+	/* 20250905
 	backlight_lcd1: backlight-lcd1 {
 		compatible = "pwm-backlight";
 		pwms = <&disp_pwm1 0 500000>;
@@ -46,11 +49,12 @@ backlight_lcd1: backlight-lcd1 {
 		num-interpolated-steps = <1023>;
 		default-brightness-level = <576>;
 	};
+	*/
 
 	chosen {
 		stdout-path = "serial0:921600n8";
 	};
-
+	
 	firmware {
 		optee {
 			compatible = "linaro,optee-tz";
@@ -60,7 +64,8 @@ optee {
 
 	memory@40000000 {
 		device_type = "memory";
-		reg = <0 0x40000000 0x1 0x00000000>;
+		//reg = <0 0x40000000 0x1 0x00000000>;
+		reg = <0 0x40000000 0x2 0x00000000>;  /* 20250905 */
 	};
 
 	reserved-memory {
@@ -151,11 +156,11 @@ edp_panel_fixed_3v3: regulator-1 {
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 		enable-active-high;
-		gpio = <&pio 15 GPIO_ACTIVE_HIGH>;
+		gpio = <&pio 111 GPIO_ACTIVE_HIGH>; /* 20250905 */
 		pinctrl-names = "default";
 		pinctrl-0 = <&edp_panel_3v3_en_pins>;
 	};
-
+	
 	gpio_fixed_3v3: regulator-2 {
 		compatible = "regulator-fixed";
 		regulator-name = "gpio_3v3_en";
@@ -175,6 +180,8 @@ sdio_fixed_1v8: regulator-3 {
 		regulator-always-on;
 	};
 
+	
+	/* 20250905
 	sdio_fixed_3v3: regulator-4 {
 		compatible = "regulator-fixed";
 		regulator-name = "sdio_card";
@@ -183,12 +190,14 @@ sdio_fixed_3v3: regulator-4 {
 		enable-active-high;
 		regulator-always-on;
 	};
+	*/
 
 	mtk_fsource: mtk-fsource {
 		compatible = "regulator-output";
 		vout-supply = <&mt6359_vefuse_ldo_reg>;
 	};
 
+	/* 20250905
 	touch0_fixed_3v3: regulator-5 {
 		compatible = "regulator-fixed";
 		regulator-name = "touch_3v3";
@@ -197,6 +206,7 @@ touch0_fixed_3v3: regulator-5 {
 		gpio = <&pio 119 GPIO_ACTIVE_HIGH>;
 		enable-active-high;
 	};
+	*/
 
 	usb_hub_fixed_3v3: regulator-6 {
 		compatible = "regulator-fixed";
@@ -217,6 +227,7 @@ usb_hub_reset_1v8: regulator-7 {
 		vin-supply = <&usb_hub_fixed_3v3>;
 	};
 
+	/* 20250905
 	usb_p0_vbus: regulator-8 {
 		compatible = "regulator-fixed";
 		regulator-name = "usb_p0_vbus";
@@ -225,7 +236,9 @@ usb_p0_vbus: regulator-8 {
 		gpio = <&pio 84 GPIO_ACTIVE_HIGH>;
 		enable-active-high;
 	};
+	*/
 
+	/* 20250905
 	usb_p1_vbus: regulator-9 {
 		compatible = "regulator-fixed";
 		regulator-name = "usb_p1_vbus";
@@ -234,7 +247,9 @@ usb_p1_vbus: regulator-9 {
 		gpio = <&pio 87 GPIO_ACTIVE_HIGH>;
 		enable-active-high;
 	};
+	*/
 
+	/* 20250905
 	usb_p2_vbus: regulator-10 {
 		compatible = "regulator-fixed";
 		regulator-name = "usb_p2_vbus";
@@ -242,15 +257,18 @@ usb_p2_vbus: regulator-10 {
 		regulator-max-microvolt = <5000000>;
 		enable-active-high;
 	};
+	*/
 
+	/* 20250905
 	dsi0_en: dsi0-en {
 		compatible = "regulator-fixed";
 		regulator-name = "lcm1_en";
 		enable-active-high;
-		gpio = <&pio 111 GPIO_ACTIVE_HIGH>;
+		gpio = <&pio 15 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&dsi0_en_pins_default>;
 	};
+	*/
 
 	amic_codec: amic-codec {
 		#sound-dai-cells = <0>;
@@ -395,10 +413,22 @@ &gpu {
 
 &i2c0 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&i2c0_pins>;
+	//pinctrl-0 = <&i2c0_pins>; /* 20250905 */
+	pinctrl-0 = <&i2c0_pins &tlv320_aud_rst>;  /* 20250905 */
 	clock-frequency = <400000>;
 	status = "okay";
+	
+	/* 20250905 */
+	tlv320aic3106: audio-codec@1b {
+		#sound-dai-cells = <0>;
+		compatible = "ti,tlv320aic3106";
+		reg = <0x1b>;
+		ai3x-micbias-vg = <1>;
+		status = "okay";
+		reset-gpios = <&pio 6 GPIO_ACTIVE_LOW>;
+ 	};
 
+	/* 20250905
 	touchscreen@5d {
 		compatible = "goodix,gt9271";
 		reg = <0x5d>;
@@ -411,6 +441,7 @@ touchscreen@5d {
 		pinctrl-names = "default";
 		pinctrl-0 = <&touch_pins>;
 	};
+	*/
 };
 
 &i2c1 {
@@ -419,6 +450,7 @@ &i2c1 {
 	clock-frequency = <400000>;
 	status = "okay";
 
+	/* 20250905
 	it5205fn: it5205fn@48 {
 		compatible = "mediatek,it5205fn";
 		reg = <0x48>;
@@ -426,6 +458,7 @@ it5205fn: it5205fn@48 {
 		svid = /bits/ 16 <0xff01>;
 		status = "okay";
 	};
+	*/
 };
 
 &i2c2 {
@@ -445,10 +478,17 @@ &i2c3 {
 &i2c4 {
 	pinctrl-names = "default", "default";
 	pinctrl-0 = <&i2c4_pins>;
-	pinctrl-1 = <&rt1715_int_pins>;
+	//pinctrl-1 = <&rt1715_int_pins>; /* 20250905 */
 	clock-frequency = <1000000>;
 	status = "okay";
 
+	/* 20250905 */
+	pcf85063: pcf85063@51 {
+    	compatible = "nxp,pcf85063";
+    	reg = <0x51>;
+	};
+	
+	/* 20250905
 	rt1715@4e {
 		compatible = "richtek,rt1715";
 		reg = <0x4e>;
@@ -498,6 +538,7 @@ ports {
 			port@0 {};
 		};
 	};
+	*/
 };
 
 &i2c5 {
@@ -512,6 +553,17 @@ &i2c6 {
 	pinctrl-0 = <&i2c6_pins>;
 	clock-frequency = <400000>;
 	status = "okay";
+	
+	/* 20250905 */
+	baseboard_eeprom: baseboard_eeprom@57 {
+		compatible = "atmel,24c02";
+		label = "AT24C02D";
+		reg = <0x57>;
+		pagesize = <8>;
+		size = <256>;
+		address-width = <8>;
+		num-addresses = <1>;
+	};
 };
 
 &mfg0 {
@@ -559,6 +611,7 @@ &mmc1 {
 	vqmmc-supply = <&mt6359_vsim1_ldo_reg>;
 };
 
+/* 20250905
 &wifi_pwrseq {
 	status = "okay";
 	pinctrl-names = "default";
@@ -567,11 +620,13 @@ &wifi_pwrseq {
 	reset-gpios = <&pio 72 1>;
 	post-power-on-delay-ms = <200>;
 };
+*/
 
 &watchdog {
 	/delete-property/ mediatek,disable-extrst;
 };
 
+/* 20250905
 &mmc2 {
 	status = "okay";
 	pinctrl-names = "default", "state_uhs", "state_eint",
@@ -596,6 +651,7 @@ &mmc2 {
 	vqmmc-supply = <&sdio_fixed_1v8>;
 	mmc-pwrseq = <&wifi_pwrseq>;
 };
+*/
 
 &mt6359_vbbck_ldo_reg {
 	regulator-always-on;
@@ -617,9 +673,11 @@ &mt6359_vgpu11_buck_reg {
 	regulator-always-on;
 };
 
+/* 20250905
 &mt6359_vmodem_buck_reg {
 	regulator-always-on;
 };
+*/
 
 &mt6359_vpa_buck_reg {
 	regulator-max-microvolt = <3100000>;
@@ -655,9 +713,11 @@ &mt6359_vsram_others_ldo_reg {
 	regulator-coupled-max-spread = <6250>;
 };
 
+/* 20250905
 &mt6359_vsram_md_ldo_reg {
 	regulator-always-on;
 };
+*/
 
 &mt6359_vufs_ldo_reg {
 	regulator-always-on;
@@ -702,7 +762,7 @@ ethernet_phy0: ethernet-phy@1 {
 };
 
 &dp_intf1 {
-	status = "disabled";
+	status = "okay";  /* 20250905 */
 	mediatek,oob-hpd;
 	ports {
 		port {
@@ -716,7 +776,7 @@ dp_intf1_out: endpoint {
 &dp_tx {
 	pinctrl-names = "default";
 	pinctrl-0 = <&dptx_pins>;
-	status = "disabled";
+	status = "okay";  /* 20250905 */
 
 	ports {
 		#address-cells = <1>;
@@ -748,11 +808,11 @@ panel@0 {
 		status = "disabled";
 		reg = <0>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&panel_pins_default>;
-		backlight = <&backlight_lcd1>;
-		reset-gpios = <&pio 25 GPIO_ACTIVE_HIGH>;
-		enable-gpios = <&pio 45 GPIO_ACTIVE_HIGH>;
-		iovcc-supply = <&dsi0_en>;
+		//pinctrl-0 = <&panel_pins_default>; /* 20250905 */
+		//backlight = <&backlight_lcd1>; /* 20250905 */
+		//reset-gpios = <&pio 25 GPIO_ACTIVE_HIGH>;  /* 20250905 */
+		enable-gpios = <&pio 8 GPIO_ACTIVE_HIGH>; /* 20250905 */
+		//iovcc-supply = <&dsi0_en>; /* 20250905 */
 		port {
 			panel0_in: endpoint {
 				remote-endpoint = <&dsi0_out>;
@@ -769,13 +829,48 @@ dsi0_out: endpoint {
 	};
 };
 
+/* 20250905
 &disp_pwm1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&disp_pwm1_pin_default>;
-	status = "disabled";
+	status = "okay";
 };
+*/
 
 &pio {
+	/* 20250905 */
+	gpio_pins: gpio-pins {
+		cn29_gpio: cn29-gpio {
+			pinmux = <PINMUX_GPIO0__FUNC_B_GPIO0>,
+				<PINMUX_GPIO1__FUNC_B_GPIO1>,
+				<PINMUX_GPIO3__FUNC_B_GPIO3>,
+				<PINMUX_GPIO5__FUNC_B_GPIO5>,
+				<PINMUX_GPIO6__FUNC_B_GPIO6>, // TLV320_RST
+				<PINMUX_GPIO7__FUNC_B_GPIO7>,
+				<PINMUX_GPIO123__FUNC_B_GPIO123>;
+			output-high;
+		};
+
+		cn30_gpio: cn30-gpio {
+			pinmux = <PINMUX_GPIO9__FUNC_B_GPIO9>,
+				<PINMUX_GPIO10__FUNC_B_GPIO10>,
+				<PINMUX_GPIO11__FUNC_B_GPIO11>,
+				<PINMUX_GPIO12__FUNC_B_GPIO12>,
+				<PINMUX_GPIO13__FUNC_B_GPIO13>,
+				// <PINMUX_GPIO24__FUNC_B_GPIO24>, TPM RST
+				<PINMUX_GPIO14__FUNC_B_GPIO14>;
+			output-high;
+		};
+	};
+	
+	/* 20250905 */
+	tlv320_aud_rst: tlv320-aud-rst {
+		pins-rst {
+			pinmux = <PINMUX_GPIO6__FUNC_B_GPIO6>;
+			output-high;
+		};
+	};
+	
 	audio_default_pins: audio-default-pins {
 		pins-cmd-dat {
 			pinmux = <PINMUX_GPIO101__FUNC_O_AUD_CLK_MOSI>,
@@ -793,9 +888,9 @@ pins-cmd-dat {
 				 <PINMUX_GPIO116__FUNC_B0_I2SO2_WS>,
 				 <PINMUX_GPIO117__FUNC_O_I2SO2_D0>,
 				 <PINMUX_GPIO118__FUNC_O_I2SO2_D1>,
-				 <PINMUX_GPIO121__FUNC_B0_PCM_CLK>,
-				 <PINMUX_GPIO122__FUNC_B0_PCM_SYNC>,
-				 <PINMUX_GPIO124__FUNC_I0_PCM_DI>,
+				 //<PINMUX_GPIO121__FUNC_B0_PCM_CLK>, /* 20250905 */
+				 //<PINMUX_GPIO122__FUNC_B0_PCM_SYNC>, /* 20250905 */
+				 //<PINMUX_GPIO124__FUNC_I0_PCM_DI>, /* 20250905 */
 				 <PINMUX_GPIO125__FUNC_O_DMIC1_CLK>,
 				 <PINMUX_GPIO126__FUNC_I0_DMIC1_DAT>,
 				 <PINMUX_GPIO128__FUNC_O_DMIC2_CLK>,
@@ -812,7 +907,8 @@ pins-cmd-dat {
 
 	edp_panel_3v3_en_pins: edp-panel-3v3-en-pins {
 		pins1 {
-			pinmux = <PINMUX_GPIO15__FUNC_B_GPIO15>;
+			//pinmux = <PINMUX_GPIO15__FUNC_B_GPIO15>; /* 20250905 */
+			pinmux = <PINMUX_GPIO111__FUNC_B_GPIO111>; /* 20250905 */
 			output-high;
 		};
 	};
@@ -838,7 +934,7 @@ pins-power {
 				 <PINMUX_GPIO146__FUNC_B_GPIO146>;
 			output-high;
 		};
-
+		
 		pins-rxd {
 			pinmux = <PINMUX_GPIO135__FUNC_I0_GBE_RXD3>,
 				 <PINMUX_GPIO136__FUNC_I0_GBE_RXD2>,
@@ -891,7 +987,8 @@ pins {
 			pinmux = <PINMUX_GPIO56__FUNC_B1_SDA0>,
 				 <PINMUX_GPIO55__FUNC_B1_SCL0>;
 			bias-pull-up = <MTK_PULL_SET_RSEL_011>;
-			drive-strength-microamp = <1000>;
+			//drive-strength-microamp = <1000>; /* 20250905 */
+			drive-strength = <MTK_DRIVE_6mA>; /* 20250905 */
 		};
 	};
 
@@ -900,7 +997,8 @@ pins {
 			pinmux = <PINMUX_GPIO58__FUNC_B1_SDA1>,
 				 <PINMUX_GPIO57__FUNC_B1_SCL1>;
 			bias-pull-up = <MTK_PULL_SET_RSEL_011>;
-			drive-strength-microamp = <1000>;
+			//drive-strength-microamp = <1000>; /* 20250905 */
+			drive-strength = <MTK_DRIVE_6mA>; /* 20250905 */
 		};
 	};
 
@@ -909,7 +1007,8 @@ pins {
 			pinmux = <PINMUX_GPIO60__FUNC_B1_SDA2>,
 				 <PINMUX_GPIO59__FUNC_B1_SCL2>;
 			bias-pull-up = <MTK_PULL_SET_RSEL_011>;
-			drive-strength-microamp = <1000>;
+			//drive-strength-microamp = <1000>; /* 20250905 */
+			drive-strength = <MTK_DRIVE_6mA>; /* 20250905 */
 		};
 	};
 
@@ -918,7 +1017,8 @@ pins {
 			pinmux = <PINMUX_GPIO62__FUNC_B1_SDA3>,
 				 <PINMUX_GPIO61__FUNC_B1_SCL3>;
 			bias-pull-up = <MTK_PULL_SET_RSEL_011>;
-			drive-strength-microamp = <1000>;
+			//drive-strength-microamp = <1000>; /* 20250905 */
+			drive-strength = <MTK_DRIVE_6mA>; /* 20250905 */
 		};
 	};
 
@@ -927,7 +1027,8 @@ pins {
 			pinmux = <PINMUX_GPIO64__FUNC_B1_SDA4>,
 				 <PINMUX_GPIO63__FUNC_B1_SCL4>;
 			bias-pull-up = <MTK_PULL_SET_RSEL_011>;
-			drive-strength-microamp = <1000>;
+			//drive-strength-microamp = <1000>; /* 20250905 */
+			drive-strength = <MTK_DRIVE_6mA>; /* 20250905 */
 		};
 	};
 
@@ -936,7 +1037,8 @@ pins {
 			pinmux = <PINMUX_GPIO66__FUNC_B1_SDA5>,
 				 <PINMUX_GPIO65__FUNC_B1_SCL5>;
 			bias-pull-up = <MTK_PULL_SET_RSEL_011>;
-			drive-strength-microamp = <1000>;
+			//drive-strength-microamp = <1000>; /* 20250905 */
+			drive-strength = <MTK_DRIVE_6mA>; /* 20250905 */
 		};
 	};
 
@@ -945,10 +1047,11 @@ pins {
 			pinmux = <PINMUX_GPIO68__FUNC_B1_SDA6>,
 				 <PINMUX_GPIO67__FUNC_B1_SCL6>;
 			bias-pull-up = <MTK_PULL_SET_RSEL_011>;
-			drive-strength-microamp = <1000>;
+			//drive-strength-microamp = <1000>; /* 20250905 */
+			drive-strength = <MTK_DRIVE_6mA>; /* 20250905 */
 		};
 	};
-
+	
 	pcie_pins_default: pcie-default {
 		mux {
 			pinmux = <PINMUX_GPIO47__FUNC_I1_WAKEN>,
@@ -966,9 +1069,10 @@ pins {
 		};
 	};
 
+	/* 20250905
 	panel_pins_default: panel-pins-default {
 		panel-dcdc {
-			pinmux = <PINMUX_GPIO45__FUNC_B_GPIO45>;
+			pinmux = <PINMUX_GPIO8__FUNC_B_GPIO8>;
 			output-low;
 		};
 
@@ -977,19 +1081,24 @@ panel-rst {
 			output-high;
 		};
 	};
+	*/
 
+	/* 20250905
 	dsi0_en_pins_default: dsi0-en-pins-default {
 		panel-en {
-			pinmux = <PINMUX_GPIO111__FUNC_B_GPIO111>;
+			pinmux = <PINMUX_GPIO15__FUNC_B_GPIO15>;
 			output-low;
 		};
 	};
+	*/
 
+	/* 20250905
 	disp_pwm1_pin_default: disp-pwm1-pin-default {
 		pins1 {
-			pinmux = <PINMUX_GPIO30__FUNC_O_DISP_PWM1>;
+			pinmux = <PINMUX_GPIO29__FUNC_O_DISP_PWM0>;
 		};
 	};
+	*/
 
 	mmc0_default_pins: mmc0-default-pins {
 		pins-clk {
@@ -1098,6 +1207,7 @@ pins-cmd-dat {
 		};
 	};
 
+	/* 20250905
 	mmc2_default_pins: mmc2-default-pins {
 		pins-clk {
 			pinmux = <PINMUX_GPIO170__FUNC_B1_MSDC2_CLK>;
@@ -1126,7 +1236,9 @@ wcn-3v3-en {
 		};
 
 	};
+	*/
 
+	/* 20250905
 	mmc2_uhs_pins: mmc2-uhs-pins {
 		pins-clk {
 			pinmux = <PINMUX_GPIO170__FUNC_B1_MSDC2_CLK>;
@@ -1145,7 +1257,9 @@ pins-cmd-dat {
 			bias-pull-up = <MTK_PUPD_SET_R1R0_01>;
 		};
 	};
+	*/
 
+	/* 20250905
 	mmc2_eint_pins: mmc2-eint-pins {
 		pins-dat1 {
 			pinmux = <PINMUX_GPIO172__FUNC_B_GPIO172>;
@@ -1153,7 +1267,9 @@ pins-dat1 {
 			bias-pull-up = <MTK_PUPD_SET_R1R0_01>;
 		};
 	};
+	*/
 
+	/* 20250905
 	mmc2_dat1_pins: mmc2-dat1-pins {
 		pins-dat1 {
 			pinmux = <PINMUX_GPIO172__FUNC_B1_MSDC2_DAT1>;
@@ -1162,7 +1278,9 @@ pins-dat1 {
 			bias-pull-up = <MTK_PUPD_SET_R1R0_01>;
 		};
 	};
+	*/
 
+	/* 20250905
 	panel_default_pins: panel-default-pins {
 		pins-dcdc {
 			pinmux = <PINMUX_GPIO45__FUNC_B_GPIO45>;
@@ -1179,7 +1297,9 @@ pins-rst {
 			output-high;
 		};
 	};
+	*/
 
+	/* 20250905
 	rt1715_int_pins: rt1715-int-pins {
 		pins_cmd0_dat {
 			pinmux = <PINMUX_GPIO12__FUNC_B_GPIO12>;
@@ -1187,6 +1307,7 @@ pins_cmd0_dat {
 			input-enable;
 		};
 	};
+	*/
 
 	spi0_pins: spi0-pins {
 		pins-spi {
@@ -1217,7 +1338,16 @@ pins-spi {
 			bias-disable;
 		};
 	};
+	
+	/* 20250905 */
+	npct75x_tpm_pin: npct75x-tpm-pin {
+		tpm_rst: tpm-rst {
+			pinmux = <PINMUX_GPIO24__FUNC_B_GPIO24>;
+			output-high;
+		};
+	};
 
+	/* 20250905
 	touch_pins: touch-pins {
 		pins-irq {
 			pinmux = <PINMUX_GPIO6__FUNC_B_GPIO6>;
@@ -1230,6 +1360,7 @@ pins-reset {
 			output-high;
 		};
 	};
+	*/
 
 	uart0_pins: uart0-pins {
 		pins {
@@ -1242,7 +1373,9 @@ pins {
 	uart1_pins: uart1-pins {
 		pins {
 			pinmux = <PINMUX_GPIO33__FUNC_O_UTXD1>,
-				 <PINMUX_GPIO34__FUNC_I1_URXD1>;
+				 <PINMUX_GPIO34__FUNC_I1_URXD1>,
+				 <PINMUX_GPIO88__FUNC_O_URTS1>,    /* 20250905 */
+				 <PINMUX_GPIO89__FUNC_I1_UCTS1>;   /* 20250905 */
 			bias-pull-up;
 		};
 	};
@@ -1250,10 +1383,20 @@ pins {
 	uart2_pins: uart2-pins {
 		pins {
 			pinmux = <PINMUX_GPIO35__FUNC_O_UTXD2>,
-				 <PINMUX_GPIO36__FUNC_I1_URXD2>;
+				 <PINMUX_GPIO36__FUNC_I1_URXD2>,
+				 <PINMUX_GPIO86__FUNC_O_URTS2>,    /* 20250905 */
+				 <PINMUX_GPIO87__FUNC_I1_UCTS2>;   /* 20250905 */
 			bias-pull-up;
 		};
 	};
+	
+	/* 20250905 */
+	uart3_pins: uart3-pins {
+		pins {
+			pinmux = <PINMUX_GPIO90__FUNC_O_UTXD3>,
+				 <PINMUX_GPIO91__FUNC_I1_URXD3>;
+		};
+	};
 
 	usb_default_pins: usb-default-pins {
 		pins-iddig {
@@ -1267,13 +1410,15 @@ pins-valid {
 			input-enable;
 		};
 
+		/* 20250905
 		pins-vbus {
 			pinmux = <PINMUX_GPIO84__FUNC_O_USB_DRVVBUS>;
 			output-high;
 		};
-
+		*/
 	};
 
+	/* 20250905
 	usb1_default_pins: usb1-default-pins {
 		pins-valid {
 			pinmux = <PINMUX_GPIO88__FUNC_I0_VBUSVALID_1P>;
@@ -1285,16 +1430,46 @@ pins-usb-hub-3v3-en {
 			output-high;
 		};
 	};
+	*/
 
+	/* 20250905
 	wifi_pwrseq_pins: wifi-pwrseq-pins {
 		pins-wifi-enable {
 			pinmux = <PINMUX_GPIO72__FUNC_B_GPIO72>;
 			output-low;
 		};
 	};
+	*/
+	
+	/* 20250905 */
+	/*
+	pwr_ctrl_gpio_pins: pwr-ctrl-gpio-pins {
+		carrier_pwr_on_gpio_pin: carrier-pwr-on-gpio-pin{
+			pinmux = <PINMUX_GPIO121__FUNC_B_GPIO121>; // AD30
+ 			output-high;
+ 		};
+
+		carrier_stby_gpio_pin: carrier-stby-gpio-pin{
+			pinmux = <PINMUX_GPIO27__FUNC_B_GPIO27>; // AA11
+ 			output-high;
+ 		};
+
+		power_supply_1v8_en_pin: power-supply-1v8-en-pin {
+			pinmux = <PINMUX_GPIO145__FUNC_B_GPIO145>;		// AC7 1.8V Power Supply
+			output-high;
+		};
+
+		power_supply_3v3_en_pin: power-supply-3v3-en-pin {
+			pinmux = <PINMUX_GPIO146__FUNC_B_GPIO146>;			// AB4 3.3V Power Supply
+			output-high;
+		};
+ 	};
+	*/
 };
 
 &pmic {
+	//pinctrl-names = "default"; /* 20250905 */
+	//pinctrl-0 = <&pwr_ctrl_gpio_pins>; /* 20250905 */
 	interrupt-parent = <&pio>;
 	interrupts = <222 IRQ_TYPE_LEVEL_HIGH>;
 };
@@ -1323,17 +1498,64 @@ &uart2 {
 	status = "okay";
 };
 
+/* 20250905 */
+&uart3 {
+	pinctrl-0 = <&uart3_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
 &mipi_tx0 {
-	status = "disabled";
+	status = "okay"; /* 20250905 */
+};
+
+/* 20250905 */
+&spi0 {
+	pinctrl-0 = <&spi0_pins>;
+	pinctrl-names = "default";
+	mediatek,pad-select = <0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	spidev@0 {
+ 		compatible = "mediatek,aiot-board";
+		spi-max-frequency = <50000000>;
+		reg = <0>;
+	};
+};
+
+/* 20250905 */
+&spi1 {
+	pinctrl-0 = <&spi1_pins>;
+	pinctrl-names = "default";
+	mediatek,pad-select = <0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	spidev@0 {
+		compatible = "mediatek,aiot-board";
+		spi-max-frequency = <50000000>;
+		reg = <0>;
+	};
 };
 
 &spi2 {
-	pinctrl-0 = <&spi2_pins>;
+	//pinctrl-0 = <&spi2_pins>; /* 20250905 */
+	pinctrl-0 = <&spi2_pins &npct75x_tpm_pin>; /* 20250905 */
 	pinctrl-names = "default";
 	mediatek,pad-select = <0>;
 	#address-cells = <1>;
 	#size-cells = <0>;
 	status = "okay";
+	
+	tpm: npct75x@0 {
+		compatible = "tcg,tpm_tis-spi";
+		reg = <0>;
+		spi-max-frequency = <1000000>;
+		reset-gpios = <&pio 24 GPIO_ACTIVE_LOW>;
+	};
 };
 
 &spmi {
@@ -1403,7 +1625,7 @@ connector {
 		compatible = "gpio-usb-b-connector", "usb-b-connector";
 		type = "micro";
 		id-gpios = <&pio 83 GPIO_ACTIVE_HIGH>;
-		vbus-supply = <&usb_p0_vbus>;
+		//vbus-supply = <&usb_p0_vbus>; /* 20250905 */
 	};
 };
 
@@ -1417,20 +1639,23 @@ &xhci0 {
 };
 
 &ssusb1 {
-	pinctrl-0 = <&usb1_default_pins>;
+	//pinctrl-0 = <&usb1_default_pins>; /* 20250905 */
 	pinctrl-names = "default";
 	vusb33-supply = <&mt6359_vusb_ldo_reg>;
-	dr_mode = "otg";
+	//dr_mode = "otg"; /* 20250905 */
+	dr_mode = "host";  /* 20250905 */
 	mediatek,usb3-drd;
 	usb-role-switch;
 	status = "okay";
 	wakeup-source;
 
+	/* 20250905
 	port {
 		ssusb1_ep: endpoint {
 			remote-endpoint = <&tcpc_typec_usb>;
 		};
 	};
+	*/
 };
 
 &u2port1 {
@@ -1444,7 +1669,7 @@ &u3port1 {
 &xhci1 {
 	status = "okay";
 	vusb33-supply = <&mt6359_vusb_ldo_reg>;
-	vbus-supply = <&usb_hub_reset_1v8>;
+	//vbus-supply = <&usb_hub_reset_1v8>; /* 20250905 */
 };
 
 &ssusb2 {
@@ -1462,7 +1687,7 @@ &u2port2 {
 
 &xhci2 {
 	status = "okay";
-	vusb33-supply = <&mt6359_vusb_ldo_reg>;
+	//vusb33-supply = <&mt6359_vusb_ldo_reg>; /* 20250905 */
 };
 
 &cam_vcore {
@@ -1478,6 +1703,7 @@ &accdet {
 };
 
 &afe {
+	mediatek,etdm-in2-cowork-source = <3>; /*20250905 */
 	#sound-dai-cells = <0>;
 	memory-region = <&afe_dma_mem_reserved>;
 	mediatek,dmic-iir-on;
@@ -1535,7 +1761,7 @@ codec {
 	dai-link-3 {
 		sound-dai = <&afe>;
 		link-name = "ETDM3_OUT_BE";
-		status = "disabled";
+		status = "okay"; /*20250905 */
 
 		codec {
 			sound-dai = <&hdmi0>;
@@ -1545,10 +1771,32 @@ codec {
 	dai-link-4 {
 		sound-dai = <&afe>;
 		link-name = "DPTX_BE";
-		status = "disabled";
+		status = "okay"; /* 20250905 */
 
 		codec {
 			sound-dai = <&dp_tx>;
 		};
 	};
+	
+	/* 20250905 */
+	dai-link-5 {
+		sound-dai = <&afe>;
+		link-name = "ETDM2_OUT_BE";
+		status = "okay";
+
+		codec {
+			sound-dai = <&tlv320aic3106>;
+		};
+	};
+	
+	/* 20250905 */
+	dai-link-6 {
+		sound-dai = <&afe>;
+		link-name = "ETDM2_IN_BE";
+		status = "okay";
+
+		codec {
+			sound-dai = <&tlv320aic3106>;
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/mediatek/mt8370.dtsi b/arch/arm64/boot/dts/mediatek/mt8370.dtsi
index 1f83a293d543..f67883aa80cf 100644
--- a/arch/arm64/boot/dts/mediatek/mt8370.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8370.dtsi
@@ -443,13 +443,15 @@ cpu-little0-thermal {
 
 			trips {
 				cpu_little0_switch_on: trip-switch-on0 {
-					temperature = <68000>;
+					//temperature = <68000>;
+					temperature = <115000>; //20250905
 					hysteresis = <2000>;
 					type = "passive";
 				};
 
 				cpu_little0_alert0: trip-alert0 {
-					temperature = <85000>;
+					//temperature = <85000>;
+					temperature = <115000>; //20250905
 					hysteresis = <2000>;
 					type = "passive";
 				};
@@ -1817,6 +1819,7 @@ eth: ethernet@11021000 {
 			reg = <0 0x11021000 0 0x4000>;
 			interrupts = <GIC_SPI 716 IRQ_TYPE_LEVEL_HIGH 0>;
 			interrupt-names = "macirq";
+			mac-address = [00 55 7b b5 7d f7];
 			clock-names = "axi",
 				      "apb",
 				      "mac_cg",
diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index 5aac741453dc..f7435349922c 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -1,3 +1,4 @@
+CONFIG_LOCALVERSION="-mtk+g778343d9829d"
 CONFIG_SYSVIPC=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_AUDIT=y
@@ -5,19 +6,23 @@ CONFIG_NO_HZ_IDLE=y
 CONFIG_HIGH_RES_TIMERS=y
 CONFIG_BPF_SYSCALL=y
 CONFIG_BPF_JIT=y
+CONFIG_BPF_LSM=y
 CONFIG_PREEMPT=y
 CONFIG_IRQ_TIME_ACCOUNTING=y
 CONFIG_BSD_PROCESS_ACCT=y
 CONFIG_BSD_PROCESS_ACCT_V3=y
 CONFIG_TASKSTATS=y
+CONFIG_TASK_DELAY_ACCT=y
 CONFIG_TASK_XACCT=y
 CONFIG_TASK_IO_ACCOUNTING=y
 CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
+CONFIG_IKHEADERS=y
 CONFIG_NUMA_BALANCING=y
 CONFIG_MEMCG=y
 CONFIG_BLK_CGROUP=y
 CONFIG_CGROUP_PIDS=y
+CONFIG_CGROUP_RDMA=y
 CONFIG_CGROUP_FREEZER=y
 CONFIG_CGROUP_HUGETLB=y
 CONFIG_CPUSETS=y
@@ -25,11 +30,15 @@ CONFIG_CGROUP_DEVICE=y
 CONFIG_CGROUP_CPUACCT=y
 CONFIG_CGROUP_PERF=y
 CONFIG_CGROUP_BPF=y
+CONFIG_CGROUP_MISC=y
 CONFIG_USER_NS=y
 CONFIG_SCHED_AUTOGROUP=y
 CONFIG_BLK_DEV_INITRD=y
 CONFIG_KALLSYMS_ALL=y
 CONFIG_PROFILING=y
+CONFIG_KEXEC=y
+CONFIG_KEXEC_FILE=y
+CONFIG_CRASH_DUMP=y
 CONFIG_ARCH_ACTIONS=y
 CONFIG_ARCH_SUNXI=y
 CONFIG_ARCH_ALPINE=y
@@ -77,9 +86,6 @@ CONFIG_ARM64_VA_BITS_48=y
 CONFIG_SCHED_MC=y
 CONFIG_SCHED_SMT=y
 CONFIG_NUMA=y
-CONFIG_KEXEC=y
-CONFIG_KEXEC_FILE=y
-CONFIG_CRASH_DUMP=y
 CONFIG_XEN=y
 CONFIG_COMPAT=y
 CONFIG_RANDOMIZE_BASE=y
@@ -100,6 +106,7 @@ CONFIG_ARM_ARMADA_37XX_CPUFREQ=y
 CONFIG_ARM_SCPI_CPUFREQ=y
 CONFIG_ARM_IMX_CPUFREQ_DT=m
 CONFIG_ARM_MEDIATEK_CPUFREQ=y
+CONFIG_ARM_MEDIATEK_CPUFREQ_HW=y
 CONFIG_ARM_QCOM_CPUFREQ_NVMEM=y
 CONFIG_ARM_QCOM_CPUFREQ_HW=y
 CONFIG_ARM_RASPBERRYPI_CPUFREQ=m
@@ -116,10 +123,10 @@ CONFIG_ACPI_APEI_MEMORY_FAILURE=y
 CONFIG_ACPI_APEI_EINJ=y
 CONFIG_VIRTUALIZATION=y
 CONFIG_KVM=y
+CONFIG_KPROBES=y
 CONFIG_JUMP_LABEL=y
 CONFIG_MODULES=y
 CONFIG_MODULE_UNLOAD=y
-CONFIG_IOSCHED_BFQ=y
 # CONFIG_CORE_DUMP_DEFAULT_ELF_HEADERS is not set
 # CONFIG_COMPAT_BRK is not set
 CONFIG_MEMORY_HOTPLUG=y
@@ -129,13 +136,15 @@ CONFIG_MEMORY_FAILURE=y
 CONFIG_TRANSPARENT_HUGEPAGE=y
 CONFIG_NET=y
 CONFIG_PACKET=y
-CONFIG_UNIX=y
-CONFIG_INET=y
+CONFIG_XDP_SOCKETS=y
+CONFIG_XDP_SOCKETS_DIAG=y
 CONFIG_IP_MULTICAST=y
 CONFIG_IP_PNP=y
 CONFIG_IP_PNP_DHCP=y
 CONFIG_IP_PNP_BOOTP=y
 CONFIG_IPV6=m
+CONFIG_IPV6_SUBTREES=y
+CONFIG_IPV6_SEG6_LWTUNNEL=y
 CONFIG_NETFILTER=y
 CONFIG_BRIDGE_NETFILTER=m
 CONFIG_NF_CONNTRACK=m
@@ -162,26 +171,33 @@ CONFIG_IP6_NF_TARGET_MASQUERADE=m
 CONFIG_BRIDGE=m
 CONFIG_BRIDGE_VLAN_FILTERING=y
 CONFIG_NET_DSA=m
-CONFIG_VLAN_8021Q=m
+CONFIG_VLAN_8021Q=y
 CONFIG_VLAN_8021Q_GVRP=y
 CONFIG_VLAN_8021Q_MVRP=y
 CONFIG_NET_SCHED=y
-CONFIG_NET_SCH_CBS=m
-CONFIG_NET_SCH_ETF=m
-CONFIG_NET_SCH_TAPRIO=m
+CONFIG_NET_SCH_MULTIQ=y
+CONFIG_NET_SCH_CBS=y
+CONFIG_NET_SCH_ETF=y
+CONFIG_NET_SCH_TAPRIO=y
 CONFIG_NET_SCH_MQPRIO=m
 CONFIG_NET_SCH_INGRESS=m
 CONFIG_NET_CLS_BASIC=m
+CONFIG_NET_CLS_U32=y
+CONFIG_NET_CLS_BPF=m
 CONFIG_NET_CLS_FLOWER=m
 CONFIG_NET_CLS_ACT=y
 CONFIG_NET_ACT_GACT=m
 CONFIG_NET_ACT_MIRRED=m
+CONFIG_NET_ACT_SKBEDIT=y
+CONFIG_NET_ACT_BPF=m
 CONFIG_NET_ACT_GATE=m
 CONFIG_QRTR_SMD=m
 CONFIG_QRTR_TUN=m
+CONFIG_CGROUP_NET_PRIO=y
+CONFIG_CGROUP_NET_CLASSID=y
+CONFIG_BPF_STREAM_PARSER=y
+CONFIG_NET_PKTGEN=y
 CONFIG_CAN=m
-CONFIG_CAN_M_CAN=m
-CONFIG_CAN_M_CAN_PLATFORM=m
 CONFIG_BT=m
 CONFIG_BT_HIDP=m
 # CONFIG_BT_LE is not set
@@ -196,9 +212,12 @@ CONFIG_BT_HCIUART_QCA=y
 CONFIG_BT_HCIUART_MRVL=y
 CONFIG_BT_MRVL=m
 CONFIG_BT_MRVL_SDIO=m
+CONFIG_BT_MTKSDIO=m
 CONFIG_BT_QCOMSMD=m
 CONFIG_BT_NXPUART=m
 CONFIG_CFG80211=m
+CONFIG_NL80211_TESTMODE=y
+CONFIG_CFG80211_WEXT=y
 CONFIG_MAC80211=m
 CONFIG_MAC80211_LEDS=y
 CONFIG_RFKILL=m
@@ -210,32 +229,33 @@ CONFIG_NFC_S3FWRN5_I2C=m
 CONFIG_PCI=y
 CONFIG_PCIEPORTBUS=y
 CONFIG_PCIEAER=y
+CONFIG_PCIEASPM_POWER_SUPERSAVE=y
 CONFIG_PCI_IOV=y
 CONFIG_PCI_PASID=y
 CONFIG_HOTPLUG_PCI=y
 CONFIG_HOTPLUG_PCI_ACPI=y
 CONFIG_PCI_AARDVARK=y
-CONFIG_PCI_TEGRA=y
-CONFIG_PCIE_RCAR_HOST=y
-CONFIG_PCIE_RCAR_EP=y
-CONFIG_PCI_HOST_GENERIC=y
-CONFIG_PCI_XGENE=y
 CONFIG_PCIE_ALTERA=y
 CONFIG_PCIE_ALTERA_MSI=y
+CONFIG_PCIE_BRCMSTB=m
 CONFIG_PCI_HOST_THUNDER_PEM=y
 CONFIG_PCI_HOST_THUNDER_ECAM=y
+CONFIG_PCI_HOST_GENERIC=y
+CONFIG_PCIE_MEDIATEK_GEN3=y
+CONFIG_PCI_TEGRA=y
+CONFIG_PCIE_RCAR_HOST=y
+CONFIG_PCIE_RCAR_EP=y
 CONFIG_PCIE_ROCKCHIP_HOST=m
-CONFIG_PCIE_MEDIATEK_GEN3=m
-CONFIG_PCIE_BRCMSTB=m
+CONFIG_PCI_XGENE=y
 CONFIG_PCI_IMX6_HOST=y
 CONFIG_PCI_LAYERSCAPE=y
 CONFIG_PCI_HISI=y
-CONFIG_PCIE_QCOM=y
-CONFIG_PCIE_ARMADA_8K=y
-CONFIG_PCIE_ROCKCHIP_DW_HOST=y
 CONFIG_PCIE_KIRIN=y
 CONFIG_PCIE_HISI_STB=y
+CONFIG_PCIE_ARMADA_8K=y
 CONFIG_PCIE_TEGRA194_HOST=m
+CONFIG_PCIE_QCOM=y
+CONFIG_PCIE_ROCKCHIP_DW_HOST=y
 CONFIG_PCIE_VISCONTI_HOST=y
 CONFIG_PCIE_LAYERSCAPE_GEN4=y
 CONFIG_PCI_ENDPOINT=y
@@ -247,11 +267,11 @@ CONFIG_FW_LOADER_USER_HELPER=y
 CONFIG_HISILICON_LPC=y
 CONFIG_TEGRA_ACONNECT=m
 CONFIG_MHI_BUS_PCI_GENERIC=m
-CONFIG_ARM_SCMI_PROTOCOL=y
 CONFIG_ARM_SCPI_PROTOCOL=y
 CONFIG_RASPBERRYPI_FIRMWARE=y
 CONFIG_INTEL_STRATIX10_SERVICE=y
 CONFIG_INTEL_STRATIX10_RSU=m
+CONFIG_MTK_ADSP_IPC=y
 CONFIG_EFI_CAPSULE_LOADER=y
 CONFIG_IMX_SCU=y
 CONFIG_IMX_SCU_PD=y
@@ -276,18 +296,16 @@ CONFIG_MTD_NAND_FSL_IFC=y
 CONFIG_MTD_NAND_QCOM=y
 CONFIG_MTD_SPI_NOR=y
 CONFIG_MTD_UBI=m
-CONFIG_UBIFS_FS=m
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_NBD=m
+CONFIG_BLK_DEV_RAM=y
 CONFIG_VIRTIO_BLK=y
 CONFIG_BLK_DEV_NVME=m
 CONFIG_QCOM_COINCELL=m
 CONFIG_QCOM_FASTRPC=m
-CONFIG_BATTERY_QCOM_BATTMGR=m
-CONFIG_UCSI_PMIC_GLINK=m
 CONFIG_SRAM=y
 CONFIG_PCI_ENDPOINT_TEST=m
-CONFIG_EEPROM_AT24=m
+CONFIG_EEPROM_AT24=y
 CONFIG_EEPROM_AT25=m
 CONFIG_UACCE=m
 # CONFIG_SCSI_PROC_FS is not set
@@ -308,7 +326,6 @@ CONFIG_AHCI_XGENE=y
 CONFIG_AHCI_QORIQ=y
 CONFIG_SATA_SIL24=y
 CONFIG_SATA_RCAR=y
-CONFIG_PATA_PLATFORM=y
 CONFIG_PATA_OF_PLATFORM=y
 CONFIG_MD=y
 CONFIG_BLK_DEV_MD=m
@@ -357,7 +374,7 @@ CONFIG_MLX5_CORE=m
 CONFIG_MLX5_CORE_EN=y
 CONFIG_QCOM_EMAC=m
 CONFIG_RMNET=m
-CONFIG_R8169=m
+CONFIG_R8169=y
 CONFIG_SH_ETH=y
 CONFIG_RAVB=y
 CONFIG_RENESAS_ETHER_SWITCH=y
@@ -365,7 +382,8 @@ CONFIG_SMC91X=y
 CONFIG_SMSC911X=y
 CONFIG_SNI_AVE=y
 CONFIG_SNI_NETSEC=y
-CONFIG_STMMAC_ETH=m
+CONFIG_STMMAC_ETH=y
+CONFIG_DWMAC_MEDIATEK=y
 CONFIG_DWMAC_TEGRA=m
 CONFIG_TI_K3_AM65_CPSW_NUSS=y
 CONFIG_QCOM_IPA=m
@@ -377,13 +395,14 @@ CONFIG_MARVELL_10G_PHY=y
 CONFIG_MICREL_PHY=y
 CONFIG_MICROSEMI_PHY=y
 CONFIG_AT803X_PHY=y
-CONFIG_REALTEK_PHY=y
 CONFIG_ROCKCHIP_PHY=y
 CONFIG_DP83867_PHY=y
 CONFIG_DP83869_PHY=m
 CONFIG_DP83TD510_PHY=y
 CONFIG_VITESSE_PHY=y
 CONFIG_CAN_FLEXCAN=m
+CONFIG_CAN_M_CAN=m
+CONFIG_CAN_M_CAN_PLATFORM=m
 CONFIG_CAN_RCAR=m
 CONFIG_CAN_RCAR_CANFD=m
 CONFIG_CAN_MCP251XFD=m
@@ -401,6 +420,7 @@ CONFIG_USB_NET_SMSC75XX=m
 CONFIG_USB_NET_SMSC95XX=m
 CONFIG_USB_NET_PLUSB=m
 CONFIG_USB_NET_MCS7830=m
+CONFIG_USB_NET_RNDIS_HOST=m
 CONFIG_ATH10K=m
 CONFIG_ATH10K_PCI=m
 CONFIG_ATH10K_SDIO=m
@@ -414,6 +434,7 @@ CONFIG_MWIFIEX=m
 CONFIG_MWIFIEX_SDIO=m
 CONFIG_MWIFIEX_PCIE=m
 CONFIG_MT7921E=m
+CONFIG_MT7921S=m
 CONFIG_RSI_91X=m
 CONFIG_WL18XX=m
 CONFIG_WLCORE_SDIO=m
@@ -426,12 +447,12 @@ CONFIG_KEYBOARD_GPIO=y
 CONFIG_KEYBOARD_SNVS_PWRKEY=m
 CONFIG_KEYBOARD_IMX_SC_KEY=m
 CONFIG_KEYBOARD_CROS_EC=y
-CONFIG_KEYBOARD_MTK_PMIC=m
 CONFIG_KEYBOARD_MT6779=y
+CONFIG_KEYBOARD_MTK_PMIC=m
 CONFIG_MOUSE_ELAN_I2C=m
 CONFIG_INPUT_TOUCHSCREEN=y
 CONFIG_TOUCHSCREEN_ATMEL_MXT=m
-CONFIG_TOUCHSCREEN_GOODIX=m
+CONFIG_TOUCHSCREEN_GOODIX=y
 CONFIG_TOUCHSCREEN_ELAN=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
 CONFIG_INPUT_MISC=y
@@ -491,12 +512,11 @@ CONFIG_HW_RANDOM=y
 CONFIG_HW_RANDOM_VIRTIO=y
 CONFIG_TCG_TPM=y
 CONFIG_TCG_TIS=m
-CONFIG_TCG_TIS_SPI=m
+CONFIG_TCG_TIS_SPI=y
 CONFIG_TCG_TIS_SPI_CR50=y
 CONFIG_TCG_TIS_I2C_CR50=m
 CONFIG_TCG_TIS_I2C_INFINEON=y
 CONFIG_I2C_CHARDEV=y
-CONFIG_I2C_MUX=y
 CONFIG_I2C_MUX_PCA954x=y
 CONFIG_I2C_BCM2835=m
 CONFIG_I2C_CADENCE=m
@@ -557,7 +577,7 @@ CONFIG_SPI_TEGRA210_QUAD=m
 CONFIG_SPI_TEGRA114=m
 CONFIG_SPI_SPIDEV=m
 CONFIG_SPMI=y
-CONFIG_SPMI_MTK_PMIF=m
+CONFIG_SPMI_MTK_PMIF=y
 CONFIG_PINCTRL_MAX77620=y
 CONFIG_PINCTRL_RK805=m
 CONFIG_PINCTRL_SINGLE=y
@@ -574,9 +594,9 @@ CONFIG_PINCTRL_IMX8DXL=y
 CONFIG_PINCTRL_IMX8ULP=y
 CONFIG_PINCTRL_IMX93=y
 CONFIG_PINCTRL_MSM=y
-CONFIG_PINCTRL_IPQ8074=y
 CONFIG_PINCTRL_IPQ5018=y
 CONFIG_PINCTRL_IPQ5332=y
+CONFIG_PINCTRL_IPQ8074=y
 CONFIG_PINCTRL_IPQ6018=y
 CONFIG_PINCTRL_IPQ9574=y
 CONFIG_PINCTRL_MSM8916=y
@@ -588,12 +608,10 @@ CONFIG_PINCTRL_MSM8998=y
 CONFIG_PINCTRL_QCM2290=y
 CONFIG_PINCTRL_QCS404=y
 CONFIG_PINCTRL_QDF2XXX=y
-CONFIG_PINCTRL_QCOM_SPMI_PMIC=y
 CONFIG_PINCTRL_QDU1000=y
 CONFIG_PINCTRL_SA8775P=y
 CONFIG_PINCTRL_SC7180=y
 CONFIG_PINCTRL_SC7280=y
-CONFIG_PINCTRL_SC7280_LPASS_LPI=m
 CONFIG_PINCTRL_SC8180X=y
 CONFIG_PINCTRL_SC8280XP=y
 CONFIG_PINCTRL_SDM660=y
@@ -605,14 +623,16 @@ CONFIG_PINCTRL_SM6350=y
 CONFIG_PINCTRL_SM6375=y
 CONFIG_PINCTRL_SM8150=y
 CONFIG_PINCTRL_SM8250=y
-CONFIG_PINCTRL_SM8250_LPASS_LPI=m
 CONFIG_PINCTRL_SM8350=y
 CONFIG_PINCTRL_SM8450=y
+CONFIG_PINCTRL_SM8550=y
+CONFIG_PINCTRL_QCOM_SPMI_PMIC=y
+CONFIG_PINCTRL_LPASS_LPI=m
+CONFIG_PINCTRL_SC7280_LPASS_LPI=m
+CONFIG_PINCTRL_SM8250_LPASS_LPI=m
 CONFIG_PINCTRL_SM8450_LPASS_LPI=m
 CONFIG_PINCTRL_SC8280XP_LPASS_LPI=m
-CONFIG_PINCTRL_SM8550=y
 CONFIG_PINCTRL_SM8550_LPASS_LPI=m
-CONFIG_PINCTRL_LPASS_LPI=m
 CONFIG_GPIO_ALTERA=m
 CONFIG_GPIO_DAVINCI=y
 CONFIG_GPIO_DWAPB=y
@@ -621,10 +641,11 @@ CONFIG_GPIO_MPC8XXX=y
 CONFIG_GPIO_MXC=y
 CONFIG_GPIO_PL061=y
 CONFIG_GPIO_RCAR=y
+CONFIG_GPIO_SYSCON=y
 CONFIG_GPIO_UNIPHIER=y
+CONFIG_GPIO_VF610=y
 CONFIG_GPIO_VISCONTI=y
 CONFIG_GPIO_WCD934X=m
-CONFIG_GPIO_VF610=y
 CONFIG_GPIO_XGENE=y
 CONFIG_GPIO_XGENE_SB=y
 CONFIG_GPIO_MAX732X=y
@@ -633,7 +654,6 @@ CONFIG_GPIO_PCA953X_IRQ=y
 CONFIG_GPIO_BD9571MWV=m
 CONFIG_GPIO_MAX77620=y
 CONFIG_GPIO_SL28CPLD=m
-CONFIG_GPIO_SYSCON=y
 CONFIG_POWER_RESET_MSM=y
 CONFIG_POWER_RESET_QCOM_PON=m
 CONFIG_POWER_RESET_XGENE=y
@@ -641,6 +661,7 @@ CONFIG_POWER_RESET_SYSCON=y
 CONFIG_POWER_RESET_SYSCON_POWEROFF=y
 CONFIG_SYSCON_REBOOT_MODE=y
 CONFIG_NVMEM_REBOOT_MODE=m
+CONFIG_BATTERY_QCOM_BATTMGR=m
 CONFIG_BATTERY_SBS=m
 CONFIG_BATTERY_BQ27XXX=y
 CONFIG_BATTERY_MAX17042=m
@@ -659,27 +680,33 @@ CONFIG_SENSORS_RASPBERRYPI_HWMON=m
 CONFIG_SENSORS_SL28CPLD=m
 CONFIG_SENSORS_INA2XX=m
 CONFIG_SENSORS_INA3221=m
+CONFIG_THERMAL_WRITABLE_TRIPS=y
+CONFIG_THERMAL_DEFAULT_GOV_POWER_ALLOCATOR=y
 CONFIG_THERMAL_GOV_POWER_ALLOCATOR=y
 CONFIG_CPU_THERMAL=y
 CONFIG_DEVFREQ_THERMAL=y
 CONFIG_THERMAL_EMULATION=y
 CONFIG_IMX_SC_THERMAL=m
 CONFIG_IMX8MM_THERMAL=m
-CONFIG_QORIQ_THERMAL=m
 CONFIG_K3_THERMAL=m
+CONFIG_QORIQ_THERMAL=m
 CONFIG_SUN8I_THERMAL=y
 CONFIG_ROCKCHIP_THERMAL=m
 CONFIG_RCAR_THERMAL=y
 CONFIG_RCAR_GEN3_THERMAL=y
 CONFIG_RZG2L_THERMAL=y
 CONFIG_ARMADA_THERMAL=y
+CONFIG_MTK_THERMAL=y
+CONFIG_MTK_SOC_THERMAL=y
+CONFIG_MTK_LVTS_THERMAL=y
+CONFIG_MTK_PMIC_THERMAL=y
 CONFIG_BCM2711_THERMAL=m
 CONFIG_BCM2835_THERMAL=m
 CONFIG_BRCMSTB_THERMAL=m
 CONFIG_EXYNOS_THERMAL=y
 CONFIG_TEGRA_SOCTHERM=m
 CONFIG_TEGRA_BPMP_THERMAL=m
-CONFIG_GENERIC_ADC_THERMAL=m
+CONFIG_GENERIC_ADC_THERMAL=y
 CONFIG_QCOM_TSENS=y
 CONFIG_QCOM_SPMI_ADC_TM5=m
 CONFIG_QCOM_SPMI_TEMP_ALARM=m
@@ -691,6 +718,7 @@ CONFIG_ARM_SP805_WATCHDOG=y
 CONFIG_ARM_SBSA_WATCHDOG=y
 CONFIG_S3C2410_WATCHDOG=y
 CONFIG_DW_WATCHDOG=y
+CONFIG_K3_RTI_WATCHDOG=m
 CONFIG_SUNXI_WATCHDOG=m
 CONFIG_NPCM7XX_WATCHDOG=y
 CONFIG_IMX2_WDT=y
@@ -706,7 +734,6 @@ CONFIG_UNIPHIER_WATCHDOG=y
 CONFIG_PM8916_WATCHDOG=m
 CONFIG_BCM2835_WDT=y
 CONFIG_BCM7038_WDT=m
-CONFIG_K3_RTI_WATCHDOG=m
 CONFIG_MFD_ALTERA_SYSMGR=y
 CONFIG_MFD_BD9571MWV=y
 CONFIG_MFD_AXP20X_I2C=y
@@ -723,8 +750,8 @@ CONFIG_MFD_RK8XX_SPI=y
 CONFIG_MFD_SEC_CORE=y
 CONFIG_MFD_SL28CPLD=y
 CONFIG_RZ_MTU3=y
-CONFIG_MFD_TPS65219=y
 CONFIG_MFD_TI_AM335X_TSCADC=m
+CONFIG_MFD_TPS65219=y
 CONFIG_MFD_ROHM_BD718XX=y
 CONFIG_MFD_WCD934X=m
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
@@ -739,7 +766,7 @@ CONFIG_REGULATOR_HI655X=y
 CONFIG_REGULATOR_MAX77620=y
 CONFIG_REGULATOR_MAX8973=y
 CONFIG_REGULATOR_MP8859=y
-CONFIG_REGULATOR_MT6315=m
+CONFIG_REGULATOR_MT6315=y
 CONFIG_REGULATOR_MT6357=y
 CONFIG_REGULATOR_MT6358=y
 CONFIG_REGULATOR_MT6359=y
@@ -759,12 +786,13 @@ CONFIG_REGULATOR_TPS65132=m
 CONFIG_REGULATOR_TPS65219=y
 CONFIG_REGULATOR_VCTRL=m
 CONFIG_RC_CORE=m
+CONFIG_LIRC=y
 CONFIG_RC_DECODERS=y
 CONFIG_RC_DEVICES=y
 CONFIG_IR_GPIO_CIR=m
 CONFIG_IR_MESON=m
 CONFIG_IR_SUNXI=m
-CONFIG_MEDIA_SUPPORT=m
+CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_MEDIA_ANALOG_TV_SUPPORT=y
 CONFIG_MEDIA_DIGITAL_TV_SUPPORT=y
@@ -777,7 +805,7 @@ CONFIG_V4L_PLATFORM_DRIVERS=y
 CONFIG_SDR_PLATFORM_DRIVERS=y
 CONFIG_V4L_MEM2MEM_DRIVERS=y
 CONFIG_VIDEO_MEDIATEK_JPEG=m
-CONFIG_VIDEO_MEDIATEK_VCODEC=m
+CONFIG_VIDEO_MEDIATEK_MDP3=m
 CONFIG_VIDEO_IMX7_CSI=m
 CONFIG_VIDEO_IMX_MIPI_CSIS=m
 CONFIG_VIDEO_IMX8_ISI=m
@@ -798,11 +826,15 @@ CONFIG_VIDEO_SAMSUNG_S5P_JPEG=m
 CONFIG_VIDEO_SAMSUNG_S5P_MFC=m
 CONFIG_VIDEO_SUN6I_CSI=m
 CONFIG_VIDEO_HANTRO=m
+CONFIG_VIDEO_AP1302=m
+CONFIG_VIDEO_IMX214=m
 CONFIG_VIDEO_IMX219=m
 CONFIG_VIDEO_IMX412=m
 CONFIG_VIDEO_OV5640=m
 CONFIG_VIDEO_OV5645=m
-CONFIG_DRM=m
+CONFIG_VIDEO_RDACM21=m
+CONFIG_VIDEO_DECODER=m
+CONFIG_DRM=y
 CONFIG_DRM_I2C_NXP_TDA998X=m
 CONFIG_DRM_HDLCD=m
 CONFIG_DRM_MALI_DISPLAY=m
@@ -825,25 +857,21 @@ CONFIG_ROCKCHIP_INNO_HDMI=y
 CONFIG_ROCKCHIP_LVDS=y
 CONFIG_DRM_RCAR_DU=m
 CONFIG_DRM_RCAR_DW_HDMI=m
-CONFIG_DRM_RCAR_MIPI_DSI=m
 CONFIG_DRM_RZG2L_MIPI_DSI=m
 CONFIG_DRM_SUN4I=m
-CONFIG_DRM_SUN6I_DSI=m
-CONFIG_DRM_SUN8I_DW_HDMI=m
-CONFIG_DRM_SUN8I_MIXER=m
 CONFIG_DRM_MSM=m
 CONFIG_DRM_TEGRA=m
 CONFIG_DRM_PANEL_BOE_TV101WUM_NL6=m
 CONFIG_DRM_PANEL_LVDS=m
 CONFIG_DRM_PANEL_SIMPLE=m
-CONFIG_DRM_PANEL_EDP=m
+CONFIG_DRM_PANEL_EDP=y
 CONFIG_DRM_PANEL_MANTIX_MLAF057WE51=m
 CONFIG_DRM_PANEL_RAYDIUM_RM67191=m
 CONFIG_DRM_PANEL_SITRONIX_ST7703=m
 CONFIG_DRM_PANEL_STARTEK_KD070FHFID015=m
+CONFIG_DRM_PANEL_STARTEK_KD070FHFID078=y
 CONFIG_DRM_PANEL_TRULY_NT35597_WQXGA=m
 CONFIG_DRM_PANEL_VISIONOX_VTDR6130=m
-CONFIG_DRM_DISPLAY_CONNECTOR=m
 CONFIG_DRM_LONTIUM_LT8912B=m
 CONFIG_DRM_LONTIUM_LT9611=m
 CONFIG_DRM_LONTIUM_LT9611UXC=m
@@ -869,19 +897,21 @@ CONFIG_DRM_VC4=m
 CONFIG_DRM_ETNAVIV=m
 CONFIG_DRM_HISI_HIBMC=m
 CONFIG_DRM_HISI_KIRIN=m
-CONFIG_DRM_MEDIATEK=m
+CONFIG_DRM_MEDIATEK=y
+CONFIG_DRM_MEDIATEK_DP=y
 CONFIG_DRM_MEDIATEK_HDMI=m
 CONFIG_DRM_MXSFB=m
-CONFIG_DRM_MESON=m
 CONFIG_DRM_IMX_LCDIF=m
+CONFIG_DRM_MESON=m
 CONFIG_DRM_PL111=m
 CONFIG_DRM_LIMA=m
 CONFIG_DRM_PANFROST=m
 CONFIG_DRM_TIDSS=m
 CONFIG_FB=y
-CONFIG_FB_MODE_HELPERS=y
 CONFIG_FB_EFI=y
-CONFIG_BACKLIGHT_PWM=m
+CONFIG_FB_MODE_HELPERS=y
+CONFIG_BACKLIGHT_CLASS_DEVICE=y
+CONFIG_BACKLIGHT_PWM=y
 CONFIG_BACKLIGHT_LP855X=m
 CONFIG_LOGO=y
 # CONFIG_LOGO_LINUX_MONO is not set
@@ -891,9 +921,11 @@ CONFIG_SND=y
 CONFIG_SND_ALOOP=m
 CONFIG_SND_HDA_TEGRA=m
 CONFIG_SND_HDA_CODEC_HDMI=m
+CONFIG_SND_USB_AUDIO=m
 CONFIG_SND_SOC=y
 CONFIG_SND_BCM2835_SOC_I2S=m
 CONFIG_SND_SOC_FSL_ASRC=m
+CONFIG_SND_SOC_FSL_SAI=y
 CONFIG_SND_SOC_FSL_MICFIL=m
 CONFIG_SND_SOC_FSL_EASRC=m
 CONFIG_SND_IMX_SOC=m
@@ -904,6 +936,8 @@ CONFIG_SND_SOC_IMX_AUDMIX=m
 CONFIG_SND_SOC_MT8183=m
 CONFIG_SND_SOC_MT8183_MT6358_TS3A227E_MAX98357A=m
 CONFIG_SND_SOC_MT8183_DA7219_MAX98357A=m
+CONFIG_SND_SOC_MT8188=y
+CONFIG_SND_SOC_MT8188_MT6359=y
 CONFIG_SND_SOC_MT8192=m
 CONFIG_SND_SOC_MT8192_MT6359_RT1015_RT5682=m
 CONFIG_SND_SOC_MT8365=m
@@ -925,6 +959,10 @@ CONFIG_SND_SOC_RK3399_GRU_SOUND=m
 CONFIG_SND_SOC_SAMSUNG=y
 CONFIG_SND_SOC_RCAR=m
 CONFIG_SND_SOC_RZ=m
+CONFIG_SND_SOC_SOF_TOPLEVEL=y
+CONFIG_SND_SOC_SOF_OF=y
+CONFIG_SND_SOC_SOF_MTK_TOPLEVEL=y
+CONFIG_SND_SOC_SOF_MT8186=m
 CONFIG_SND_SUN8I_CODEC=m
 CONFIG_SND_SUN8I_CODEC_ANALOG=m
 CONFIG_SND_SUN50I_CODEC_ANALOG=m
@@ -944,7 +982,7 @@ CONFIG_SND_SOC_TEGRA210_AMX=m
 CONFIG_SND_SOC_TEGRA210_ADX=m
 CONFIG_SND_SOC_TEGRA210_MIXER=m
 CONFIG_SND_SOC_TEGRA_AUDIO_GRAPH_CARD=m
-CONFIG_SND_SOC_DAVINCI_MCASP=m
+CONFIG_SND_SOC_J721E_EVM=m
 CONFIG_SND_SOC_AK4613=m
 CONFIG_SND_SOC_DA7213=m
 CONFIG_SND_SOC_ES7134=m
@@ -953,17 +991,17 @@ CONFIG_SND_SOC_ES8316=m
 CONFIG_SND_SOC_GTM601=m
 CONFIG_SND_SOC_MSM8916_WCD_ANALOG=m
 CONFIG_SND_SOC_MSM8916_WCD_DIGITAL=m
-CONFIG_SND_SOC_PCM3168A_I2C=m
 CONFIG_SND_SOC_RK817=m
 CONFIG_SND_SOC_RT5640=m
-CONFIG_SND_SOC_J721E_EVM=m
 CONFIG_SND_SOC_RT5659=m
 CONFIG_SND_SOC_SIMPLE_AMPLIFIER=m
 CONFIG_SND_SOC_SIMPLE_MUX=m
 CONFIG_SND_SOC_TAS2552=m
 CONFIG_SND_SOC_TAS571X=m
+CONFIG_SND_SOC_TLV320AIC31XX=y
 CONFIG_SND_SOC_TLV320AIC32X4_I2C=m
-CONFIG_SND_SOC_TLV320AIC3X_I2C=m
+CONFIG_SND_SOC_TLV320AIC3X_I2C=y
+CONFIG_SND_SOC_TLV320AIC3X_SPI=y
 CONFIG_SND_SOC_WCD9335=m
 CONFIG_SND_SOC_WCD934X=m
 CONFIG_SND_SOC_WM8524=m
@@ -975,12 +1013,13 @@ CONFIG_SND_SOC_WSA881X=m
 CONFIG_SND_SOC_NAU8822=m
 CONFIG_SND_SOC_LPASS_WSA_MACRO=m
 CONFIG_SND_SOC_LPASS_VA_MACRO=m
-CONFIG_SND_SOC_LPASS_RX_MACRO=m
-CONFIG_SND_SOC_LPASS_TX_MACRO=m
 CONFIG_SND_SIMPLE_CARD=m
 CONFIG_SND_AUDIO_GRAPH_CARD=m
 CONFIG_SND_AUDIO_GRAPH_CARD2=m
+CONFIG_UHID=y
 CONFIG_HID_MULTITOUCH=m
+CONFIG_HID_BPF=y
+CONFIG_USB_HIDDEV=y
 CONFIG_I2C_HID_ACPI=m
 CONFIG_I2C_HID_OF=m
 CONFIG_USB=y
@@ -1013,7 +1052,7 @@ CONFIG_USB_CHIPIDEA=y
 CONFIG_USB_CHIPIDEA_UDC=y
 CONFIG_USB_CHIPIDEA_HOST=y
 CONFIG_USB_ISP1760=y
-CONFIG_USB_SERIAL=m
+CONFIG_USB_SERIAL=y
 CONFIG_USB_SERIAL_CP210X=m
 CONFIG_USB_SERIAL_FTDI_SIO=m
 CONFIG_USB_SERIAL_OPTION=m
@@ -1026,7 +1065,7 @@ CONFIG_USB_RENESAS_USBHS_UDC=m
 CONFIG_USB_RZV2M_USB3DRD=y
 CONFIG_USB_RENESAS_USB3=m
 CONFIG_USB_TEGRA_XUDC=m
-CONFIG_USB_CONFIGFS=m
+CONFIG_USB_CONFIGFS=y
 CONFIG_USB_CONFIGFS_SERIAL=y
 CONFIG_USB_CONFIGFS_ACM=y
 CONFIG_USB_CONFIGFS_OBEX=y
@@ -1037,14 +1076,19 @@ CONFIG_USB_CONFIGFS_RNDIS=y
 CONFIG_USB_CONFIGFS_EEM=y
 CONFIG_USB_CONFIGFS_MASS_STORAGE=y
 CONFIG_USB_CONFIGFS_F_FS=y
+CONFIG_USB_CONFIGFS_F_UVC=y
 CONFIG_TYPEC=m
 CONFIG_TYPEC_TCPM=m
 CONFIG_TYPEC_TCPCI=m
+CONFIG_TYPEC_RT1711H=m
 CONFIG_TYPEC_FUSB302=m
-CONFIG_TYPEC_TPS6598X=m
-CONFIG_TYPEC_HD3SS3220=m
 CONFIG_TYPEC_UCSI=m
 CONFIG_UCSI_CCG=m
+CONFIG_UCSI_PMIC_GLINK=m
+CONFIG_TYPEC_TPS6598X=m
+CONFIG_TYPEC_HD3SS3220=m
+CONFIG_TYPEC_MUX_IT5205=m
+CONFIG_TYPEC_DP_ALTMODE=m
 CONFIG_MMC=y
 CONFIG_MMC_BLOCK_MINORS=32
 CONFIG_MMC_ARMMMCI=y
@@ -1108,7 +1152,7 @@ CONFIG_RTC_DRV_HYM8563=m
 CONFIG_RTC_DRV_MAX77686=y
 CONFIG_RTC_DRV_RK808=m
 CONFIG_RTC_DRV_ISL1208=m
-CONFIG_RTC_DRV_PCF85063=m
+CONFIG_RTC_DRV_PCF85063=y
 CONFIG_RTC_DRV_PCF85363=m
 CONFIG_RTC_DRV_M41T80=m
 CONFIG_RTC_DRV_BQ32K=m
@@ -1130,7 +1174,7 @@ CONFIG_RTC_DRV_TEGRA=y
 CONFIG_RTC_DRV_SNVS=m
 CONFIG_RTC_DRV_BBNSM=m
 CONFIG_RTC_DRV_IMX_SC=m
-CONFIG_RTC_DRV_MT6397=m
+CONFIG_RTC_DRV_MT6397=y
 CONFIG_RTC_DRV_XGENE=y
 CONFIG_RTC_DRV_TI_K3=m
 CONFIG_DMADEVICES=y
@@ -1146,6 +1190,7 @@ CONFIG_PL330_DMA=y
 CONFIG_TEGRA186_GPC_DMA=y
 CONFIG_TEGRA20_APB_DMA=y
 CONFIG_TEGRA210_ADMA=m
+CONFIG_MTK_UART_APDMA=y
 CONFIG_QCOM_BAM_DMA=y
 CONFIG_QCOM_GPI_DMA=m
 CONFIG_QCOM_HIDMA_MGMT=y
@@ -1155,6 +1200,9 @@ CONFIG_RENESAS_USB_DMAC=m
 CONFIG_RZ_DMAC=y
 CONFIG_TI_K3_UDMA=y
 CONFIG_TI_K3_UDMA_GLUE_LAYER=y
+CONFIG_DMABUF_HEAPS=y
+CONFIG_DMABUF_HEAPS_SYSTEM=y
+CONFIG_DMABUF_HEAPS_CMA=y
 CONFIG_VFIO=y
 CONFIG_VFIO_PCI=y
 CONFIG_VIRTIO_PCI=y
@@ -1172,7 +1220,6 @@ CONFIG_CROS_EC_RPMSG=m
 CONFIG_CROS_EC_SPI=y
 CONFIG_CROS_EC_CHARDEV=m
 CONFIG_COMMON_CLK_RK808=y
-CONFIG_COMMON_CLK_SCMI=y
 CONFIG_COMMON_CLK_SCPI=y
 CONFIG_COMMON_CLK_CS2000_CP=y
 CONFIG_COMMON_CLK_FSL_SAI=y
@@ -1190,18 +1237,6 @@ CONFIG_CLK_IMX8QXP=y
 CONFIG_CLK_IMX8ULP=y
 CONFIG_CLK_IMX93=y
 CONFIG_TI_SCI_CLK=y
-CONFIG_COMMON_CLK_MT8192_AUDSYS=y
-CONFIG_COMMON_CLK_MT8192_CAMSYS=y
-CONFIG_COMMON_CLK_MT8192_IMGSYS=y
-CONFIG_COMMON_CLK_MT8192_IMP_IIC_WRAP=y
-CONFIG_COMMON_CLK_MT8192_IPESYS=y
-CONFIG_COMMON_CLK_MT8192_MDPSYS=y
-CONFIG_COMMON_CLK_MT8192_MFGCFG=y
-CONFIG_COMMON_CLK_MT8192_MMSYS=y
-CONFIG_COMMON_CLK_MT8192_MSDC=y
-CONFIG_COMMON_CLK_MT8192_SCP_ADSP=y
-CONFIG_COMMON_CLK_MT8192_VDECSYS=y
-CONFIG_COMMON_CLK_MT8192_VENCSYS=y
 CONFIG_COMMON_CLK_QCOM=y
 CONFIG_QCOM_A53PLL=y
 CONFIG_QCOM_CLK_APCS_MSM8916=y
@@ -1209,24 +1244,23 @@ CONFIG_QCOM_CLK_APCC_MSM8996=y
 CONFIG_QCOM_CLK_SMD_RPM=y
 CONFIG_QCOM_CLK_RPMH=y
 CONFIG_IPQ_APSS_6018=y
-CONFIG_IPQ_GCC_5332=y
-CONFIG_IPQ_APSS_5018=y
 CONFIG_IPQ_GCC_5018=y
+CONFIG_IPQ_GCC_5332=y
 CONFIG_IPQ_GCC_6018=y
 CONFIG_IPQ_GCC_8074=y
 CONFIG_IPQ_GCC_9574=y
 CONFIG_MSM_GCC_8916=y
+CONFIG_MSM_MMCC_8994=m
 CONFIG_MSM_GCC_8994=y
 CONFIG_MSM_GCC_8996=y
-CONFIG_MSM_MMCC_8994=m
 CONFIG_MSM_MMCC_8996=m
-CONFIG_MSM_MMCC_8998=m
 CONFIG_MSM_GCC_8998=y
+CONFIG_MSM_MMCC_8998=m
 CONFIG_QCM_GCC_2290=y
 CONFIG_QCM_DISPCC_2290=m
 CONFIG_QCS_GCC_404=y
-CONFIG_SA_GCC_8775P=y
 CONFIG_SC_DISPCC_8280XP=m
+CONFIG_SA_GCC_8775P=y
 CONFIG_SA_GPUCC_8775P=m
 CONFIG_SC_GCC_7180=y
 CONFIG_SC_GCC_7280=y
@@ -1248,10 +1282,10 @@ CONFIG_SM_GCC_6115=y
 CONFIG_SM_GCC_8350=y
 CONFIG_SM_GCC_8450=y
 CONFIG_SM_GCC_8550=y
-CONFIG_SM_TCSRCC_8550=y
 CONFIG_SM_GPUCC_6115=m
 CONFIG_SM_GPUCC_8150=y
 CONFIG_SM_GPUCC_8250=y
+CONFIG_SM_TCSRCC_8550=y
 CONFIG_SM_VIDEOCC_8250=y
 CONFIG_QCOM_HFPLL=y
 CONFIG_CLK_GFM_LPASS_SM8250=m
@@ -1262,10 +1296,10 @@ CONFIG_TEGRA186_TIMER=y
 CONFIG_RENESAS_OSTM=y
 CONFIG_ARM_MHU=y
 CONFIG_IMX_MBOX=y
-CONFIG_OMAP2PLUS_MBOX=m
 CONFIG_PLATFORM_MHU=y
 CONFIG_BCM2835_MBOX=y
 CONFIG_QCOM_APCS_IPC=y
+CONFIG_MTK_ADSP_MBOX=y
 CONFIG_QCOM_IPCC=y
 CONFIG_ROCKCHIP_IOMMU=y
 CONFIG_TEGRA_IOMMU_SMMU=y
@@ -1275,14 +1309,14 @@ CONFIG_MTK_IOMMU=y
 CONFIG_QCOM_IOMMU=y
 CONFIG_REMOTEPROC=y
 CONFIG_IMX_REMOTEPROC=y
-CONFIG_TI_K3_R5_REMOTEPROC=m
-CONFIG_TI_K3_DSP_REMOTEPROC=m
 CONFIG_MTK_SCP=m
 CONFIG_QCOM_Q6V5_ADSP=m
 CONFIG_QCOM_Q6V5_MSS=m
 CONFIG_QCOM_Q6V5_PAS=m
 CONFIG_QCOM_SYSMON=m
 CONFIG_QCOM_WCNSS_PIL=m
+CONFIG_TI_K3_DSP_REMOTEPROC=m
+CONFIG_TI_K3_R5_REMOTEPROC=m
 CONFIG_RPMSG_CHAR=m
 CONFIG_RPMSG_CTRL=m
 CONFIG_RPMSG_QCOM_GLINK_RPM=y
@@ -1296,7 +1330,7 @@ CONFIG_RASPBERRYPI_POWER=y
 CONFIG_FSL_DPAA=y
 CONFIG_FSL_MC_DPIO=y
 CONFIG_FSL_RCPM=y
-CONFIG_MTK_CMDQ=m
+CONFIG_MTK_CMDQ=y
 CONFIG_MTK_DEVAPC=m
 CONFIG_MTK_PMIC_WRAP=y
 CONFIG_MTK_SVS=m
@@ -1349,6 +1383,8 @@ CONFIG_ARCH_TEGRA_194_SOC=y
 CONFIG_ARCH_TEGRA_234_SOC=y
 CONFIG_TI_SCI_PM_DOMAINS=y
 CONFIG_TI_PRUSS=m
+CONFIG_DEVFREQ_GOV_PERFORMANCE=y
+CONFIG_DEVFREQ_GOV_POWERSAVE=y
 CONFIG_ARM_IMX_BUS_DEVFREQ=y
 CONFIG_ARM_IMX8M_DDRC_DEVFREQ=m
 CONFIG_ARM_MEDIATEK_CCI_DEVFREQ=m
@@ -1360,7 +1396,8 @@ CONFIG_IIO=y
 CONFIG_EXYNOS_ADC=y
 CONFIG_IMX93_ADC=m
 CONFIG_MAX9611=m
-CONFIG_MEDIATEK_MT6577_AUXADC=m
+CONFIG_MEDIATEK_MT635X_AUXADC=y
+CONFIG_MEDIATEK_MT6577_AUXADC=y
 CONFIG_QCOM_SPMI_VADC=m
 CONFIG_QCOM_SPMI_ADC5=m
 CONFIG_ROCKCHIP_SARADC=m
@@ -1382,7 +1419,7 @@ CONFIG_PWM_BRCMSTB=m
 CONFIG_PWM_CROS_EC=m
 CONFIG_PWM_IMX27=m
 CONFIG_PWM_MESON=m
-CONFIG_PWM_MTK_DISP=m
+CONFIG_PWM_MTK_DISP=y
 CONFIG_PWM_MEDIATEK=m
 CONFIG_PWM_RCAR=m
 CONFIG_PWM_RENESAS_TPU=m
@@ -1403,6 +1440,7 @@ CONFIG_RESET_QCOM_AOSS=y
 CONFIG_RESET_QCOM_PDC=m
 CONFIG_RESET_RZG2L_USBPHY_CTRL=y
 CONFIG_RESET_TI_SCI=y
+CONFIG_RESET_TI_SYSCON=y
 CONFIG_PHY_XGENE=y
 CONFIG_PHY_CAN_TRANSCEIVER=m
 CONFIG_PHY_SUN4I_USB=y
@@ -1416,11 +1454,11 @@ CONFIG_PHY_HISI_INNO_USB2=y
 CONFIG_PHY_MVEBU_CP110_COMPHY=y
 CONFIG_PHY_MTK_TPHY=y
 CONFIG_PHY_QCOM_EDP=m
-CONFIG_PHY_QCOM_EUSB2_REPEATER=m
 CONFIG_PHY_QCOM_PCIE2=m
 CONFIG_PHY_QCOM_QMP=m
 CONFIG_PHY_QCOM_QUSB2=m
 CONFIG_PHY_QCOM_SNPS_EUSB2=m
+CONFIG_PHY_QCOM_EUSB2_REPEATER=m
 CONFIG_PHY_QCOM_USB_HS=m
 CONFIG_PHY_QCOM_USB_SNPS_FEMTO_V2=m
 CONFIG_PHY_QCOM_USB_HS_28NM=m
@@ -1447,7 +1485,6 @@ CONFIG_PHY_J721E_WIZ=m
 CONFIG_ARM_CCI_PMU=m
 CONFIG_ARM_CCN=m
 CONFIG_ARM_CMN=m
-CONFIG_ARM_CORESIGHT_PMU_ARCH_SYSTEM_PMU=m
 CONFIG_ARM_SMMU_V3_PMU=m
 CONFIG_ARM_DSU_PMU=m
 CONFIG_FSL_IMX8_DDR_PMU=m
@@ -1456,6 +1493,7 @@ CONFIG_QCOM_L3_PMU=y
 CONFIG_ARM_SPE_PMU=m
 CONFIG_ARM_DMC620_PMU=m
 CONFIG_HISI_PMU=y
+CONFIG_ARM_CORESIGHT_PMU_ARCH_SYSTEM_PMU=m
 CONFIG_NVMEM_LAYOUT_SL28_VPD=m
 CONFIG_NVMEM_IMX_OCOTP=y
 CONFIG_NVMEM_IMX_OCOTP_ELE=m
@@ -1463,6 +1501,7 @@ CONFIG_NVMEM_IMX_OCOTP_SCU=y
 CONFIG_NVMEM_LAYERSCAPE_SFP=m
 CONFIG_NVMEM_MESON_EFUSE=m
 CONFIG_NVMEM_MTK_EFUSE=y
+CONFIG_MT635X_EFUSE=y
 CONFIG_NVMEM_QCOM_QFPROM=y
 CONFIG_NVMEM_RMEM=m
 CONFIG_NVMEM_ROCKCHIP_EFUSE=y
@@ -1482,10 +1521,8 @@ CONFIG_TEE=y
 CONFIG_OPTEE=y
 CONFIG_MUX_GPIO=m
 CONFIG_MUX_MMIO=y
-CONFIG_SLIMBUS=m
 CONFIG_SLIM_QCOM_CTRL=m
 CONFIG_SLIM_QCOM_NGD_CTRL=m
-CONFIG_INTERCONNECT=y
 CONFIG_INTERCONNECT_IMX=y
 CONFIG_INTERCONNECT_IMX8MM=m
 CONFIG_INTERCONNECT_IMX8MN=m
@@ -1524,22 +1561,26 @@ CONFIG_QUOTA=y
 CONFIG_AUTOFS_FS=y
 CONFIG_FUSE_FS=m
 CONFIG_CUSE=m
-CONFIG_OVERLAY_FS=m
+CONFIG_OVERLAY_FS=y
 CONFIG_VFAT_FS=y
+CONFIG_EXFAT_FS=m
 CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_HUGETLBFS=y
-CONFIG_CONFIGFS_FS=y
 CONFIG_EFIVAR_FS=y
+CONFIG_UBIFS_FS=m
 CONFIG_SQUASHFS=y
 CONFIG_NFS_FS=y
 CONFIG_NFS_V4=y
 CONFIG_NFS_V4_1=y
 CONFIG_NFS_V4_2=y
 CONFIG_ROOT_NFS=y
+CONFIG_NFSD=y
 CONFIG_9P_FS=y
 CONFIG_NLS_CODEPAGE_437=y
 CONFIG_NLS_ISO8859_1=y
 CONFIG_SECURITY=y
+CONFIG_SECURITY_NETWORK=y
+CONFIG_SECURITY_PATH=y
 CONFIG_CRYPTO_USER=y
 CONFIG_CRYPTO_TEST=m
 CONFIG_CRYPTO_ECHAINIV=y
@@ -1573,12 +1614,16 @@ CONFIG_CMA_SIZE_MBYTES=32
 CONFIG_PRINTK_TIME=y
 CONFIG_DEBUG_KERNEL=y
 CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT=y
-CONFIG_DEBUG_INFO_REDUCED=y
+CONFIG_DEBUG_INFO_BTF=y
 CONFIG_MAGIC_SYSRQ=y
-CONFIG_DEBUG_FS=y
 # CONFIG_SCHED_DEBUG is not set
-# CONFIG_DEBUG_PREEMPT is not set
-# CONFIG_FTRACE is not set
+CONFIG_FUNCTION_TRACER=y
+CONFIG_FUNCTION_GRAPH_RETVAL=y
+CONFIG_FUNCTION_PROFILER=y
+CONFIG_FTRACE_SYSCALLS=y
+CONFIG_BLK_DEV_IO_TRACE=y
+CONFIG_BPF_KPROBE_OVERRIDE=y
+CONFIG_USER_EVENTS=y
 CONFIG_CORESIGHT=m
 CONFIG_CORESIGHT_LINK_AND_SINK_TMC=m
 CONFIG_CORESIGHT_CATU=m
@@ -1587,4 +1632,5 @@ CONFIG_CORESIGHT_SINK_ETBV10=m
 CONFIG_CORESIGHT_STM=m
 CONFIG_CORESIGHT_CPU_DEBUG=m
 CONFIG_CORESIGHT_CTI=m
+CONFIG_FUNCTION_ERROR_INJECTION=y
 CONFIG_MEMTEST=y
diff --git a/drivers/pmdomain/mediatek/mtk-pm-domains.c b/drivers/pmdomain/mediatek/mtk-pm-domains.c
index 1285f4c8a574..d0c5d3ed0bed 100644
--- a/drivers/pmdomain/mediatek/mtk-pm-domains.c
+++ b/drivers/pmdomain/mediatek/mtk-pm-domains.c
@@ -176,9 +176,8 @@ static int scpsys_clamp_bus_protection_enable(struct scpsys_domain *pd, bool is_
 {
 	int smi_count = 0;
 
-	if (pd->num_smi == 0) {
+	if (is_smi && pd->num_smi == 0)
 		return 0;
-	}
 
 	for (int i = 0; i < SPM_MAX_BUS_PROT_DATA; i++) {
 		const struct scpsys_bus_prot_data *bpd = &pd->data->bp_cfg[i];
@@ -213,9 +212,8 @@ static int scpsys_clamp_bus_protection_disable(struct scpsys_domain *pd, bool is
 {
 	int smi_count = pd->num_smi - 1;
 
-	if (pd->num_smi == 0) {
+	if (is_smi && pd->num_smi == 0)
 		return 0;
-	}
 
 	for (int i = SPM_MAX_BUS_PROT_DATA - 1; i >= 0; i--) {
 		const struct scpsys_bus_prot_data *bpd = &pd->data->bp_cfg[i];
@@ -509,7 +507,6 @@ generic_pm_domain *scpsys_add_one_domain(struct scpsys *scpsys, struct device_no
 		}
 	} else {
 		pd->num_smi = 0;
-		dev_info(scpsys->dev, "dts not set, skip smi clamp protection.\n");
 	}
 
 	pd->num_larb = of_count_phandle_with_args(node, "mediatek,larb", NULL);
diff --git a/sound/soc/mediatek/mt8188/mt8188-mt6359.c b/sound/soc/mediatek/mt8188/mt8188-mt6359.c
index 6d2e7641e9dd..6f38bd2dcf73 100644
--- a/sound/soc/mediatek/mt8188/mt8188-mt6359.c
+++ b/sound/soc/mediatek/mt8188/mt8188-mt6359.c
@@ -14,6 +14,8 @@
 #include <sound/jack.h>
 #include <sound/pcm_params.h>
 #include <sound/soc.h>
+#include <linux/clk.h> //20250905
+#include <sound/simple_card_utils.h> //20250905
 #include "mt8188-afe-common.h"
 #include "../../codecs/nau8825.h"
 #include "../../codecs/mt6359.h"
@@ -56,6 +58,8 @@
 
 #define RT5682S_CODEC_DAI     "rt5682s-aif1"
 
+#define TLV320AIC_CODEC_DAI	"tlv320aic3x-hifi" //20250905
+
 /* FE */
 SND_SOC_DAILINK_DEFS(playback2,
 		     DAILINK_COMP_ARRAY(COMP_CPU("DL2")),
@@ -354,6 +358,53 @@ static const struct snd_soc_dapm_route mt8188_mt6359_routes[] = {
 	{"I021", NULL, SOF_DMA_DL3},
 };
 
+//20250905
+int asoc_simple_dai_init_for_tlv320(struct snd_soc_pcm_runtime *rtd)
+{
+	struct asoc_simple_priv *priv = snd_soc_card_get_drvdata(rtd->card);
+	struct simple_dai_props *props = simple_priv_to_props(priv, rtd->num);
+	struct asoc_simple_dai *dai;
+	int i, ret;
+
+	return 0;
+}
+
+static int mt8188_etdm_hw_params_for_tlv320(struct snd_pcm_substream *substream,
+				 struct snd_pcm_hw_params *params)
+{
+	struct snd_soc_pcm_runtime *rtd = substream->private_data;
+	struct snd_soc_card *card = rtd->card;
+	struct snd_soc_dai *cpu_dai = asoc_rtd_to_cpu(rtd, 0);
+	unsigned int rate = params_rate(params);
+	unsigned int mclk_fs_ratio = 128;
+	unsigned int mclk_fs = rate * mclk_fs_ratio;
+	int ret;
+
+	ret = snd_soc_dai_set_sysclk(cpu_dai, 0, mclk_fs, SND_SOC_CLOCK_OUT);
+	if (ret) {
+		dev_err(card->dev, "failed to set sysclk\n");
+		return ret;
+	}
+
+	if (mclk_fs) {
+		struct snd_soc_dai *sdai;
+		int i=0;
+		for_each_rtd_codec_dais(rtd, i, sdai) {
+			if (strcmp(sdai->name, TLV320AIC_CODEC_DAI) == 0) {
+				ret = snd_soc_dai_set_sysclk(sdai, 0, mclk_fs, SND_SOC_CLOCK_IN);
+				if (ret && ret != -ENOTSUPP)
+					return ret;
+			}
+		}
+	}
+	
+	return 0;
+}
+
+static const struct snd_soc_ops mt8188_etdm_ops_for_tlv320 = {
+	.hw_params	= mt8188_etdm_hw_params_for_tlv320,
+};
+
 static int mt8188_mt6359_mtkaif_calibration(struct snd_soc_pcm_runtime *rtd)
 {
 	struct snd_soc_component *cmpnt_afe =
@@ -1235,7 +1286,8 @@ static struct snd_soc_dai_link mt8188_mt6359_dai_links[] = {
 			SND_SOC_DAIFMT_CBP_CFP,
 		.dpcm_capture = 1,
 		.ignore_suspend = 1,
-		.ops = &mt8188_etdm_ops,
+		//.ops = &mt8188_etdm_ops,
+		.ops = &mt8188_etdm_ops_for_tlv320, //20250905
 		SND_SOC_DAILINK_REG(etdm1_in),
 	},
 	[DAI_LINK_ETDM2_IN_BE] = {
@@ -1245,7 +1297,8 @@ static struct snd_soc_dai_link mt8188_mt6359_dai_links[] = {
 			SND_SOC_DAIFMT_NB_NF |
 			SND_SOC_DAIFMT_CBP_CFP,
 		.dpcm_capture = 1,
-		.ops = &mt8188_etdm_ops,
+		//.ops = &mt8188_etdm_ops,
+		.ops = &mt8188_etdm_ops_for_tlv320, //20250905
 		SND_SOC_DAILINK_REG(etdm2_in),
 	},
 	[DAI_LINK_ETDM1_OUT_BE] = {
@@ -1263,9 +1316,10 @@ static struct snd_soc_dai_link mt8188_mt6359_dai_links[] = {
 		.no_pcm = 1,
 		.dai_fmt = SND_SOC_DAIFMT_I2S |
 			SND_SOC_DAIFMT_NB_NF |
-			SND_SOC_DAIFMT_CBC_CFC,
+			SND_SOC_DAIFMT_CBP_CFP,
 		.dpcm_playback = 1,
-		.ops = &mt8188_etdm_ops,
+		//.ops = &mt8188_etdm_ops,
+		.ops = &mt8188_etdm_ops_for_tlv320, //20250905
 		SND_SOC_DAILINK_REG(etdm2_out),
 	},
 	[DAI_LINK_ETDM3_OUT_BE] = {
@@ -1377,6 +1431,7 @@ static int mt8188_mt6359_dev_probe(struct platform_device *pdev)
 	bool init_rt5682s = false;
 	bool init_max98390 = false;
 	bool init_dumb = false;
+	bool init_tlv320 = false;
 	int ret, i;
 
 	card_data = (struct mt8188_card_data *)of_device_get_match_data(&pdev->dev);
@@ -1498,6 +1553,12 @@ static int mt8188_mt6359_dev_probe(struct platform_device *pdev)
 					dai_link->exit = mt8188_rt5682s_codec_exit;
 					init_rt5682s = true;
 				}
+			} else if (!strcmp(dai_link->codecs->dai_name, TLV320AIC_CODEC_DAI)) {
+				dai_link->ops = &mt8188_etdm_ops_for_tlv320;
+				if (!init_tlv320) {
+					dai_link->init = asoc_simple_dai_init_for_tlv320;
+					init_tlv320 = true;
+				}
 			} else {
 				if (strcmp(dai_link->codecs->dai_name, "snd-soc-dummy-dai")) {
 					if (!init_dumb) {
